tools/src/keygen/KeygenOptions.hs

```diff
@@ -67,19 +77,43 @@ data FakeAvvmOptions = FakeAvvmOptions
     , faoOneStake :: Word64
     } deriving (Show)
 
-optionsParser :: Parser KeygenOptions
-optionsParser = do
-    koRearrangeMask <- optional $ strOption $
-        long    "rearrange-mask" <>
+keygenCommandParser :: Parser KeygenCommand
+keygenCommandParser =
+    subparser $ mconcat $
+    [ command "rearrange"
+      (infoH rearrangeMask (progDesc "Rearrange keyfiles"))
+    , command "dump-dev-keys"
+      (infoH dumpKeys (progDesc "Dump CSL dev-mode keys"))
+    , command "generate-avvm-seeds"
+      (infoH (fmap DumpAvvmSeeds dumpAvvmSeedsParser)
+            (progDesc "Generate avvm seeds with public keys"))
```

Comment:
`progDesc` is a full sentence, so I think you should add a dot at the end of it. ;-) And in all such cases too.